pkg/gitproviders/provider_org.go

```diff
@@ -125,3 +125,26 @@ func (p orgGitProvider) GetCommits(ctx context.Context, repoUrl RepoURL, targetB
 func (p orgGitProvider) GetProviderDomain() string {
 	return getProviderDomain(p.provider.ProviderID())
 }
+
+// GetRepoFiles returns the files found in a directory. The targetPath must point to a directory, not a file.
+// Note that the current implementation only gets an end subdirectory. It does not get multiple directories recursively. See https://github.com/fluxcd/go-git-providers/issues/143.
+func (p orgGitProvider) GetRepoFiles(ctx context.Context, repoUrl RepoURL, targetPath, targetBranch string) ([]*gitprovider.CommitFile, error) {
+	repo, err := p.getOrgRepo(ctx, repoUrl)
+	if err != nil {
+		return nil, err
+	}
+	files, err := repo.Files().Get(ctx, targetPath, targetBranch)
+	if err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
+// MergePullRequest merges a pull request given the repository's URL and the PR's number with a merge method, and a commit message.
+func (p orgGitProvider) MergePullRequest(ctx context.Context, repoUrl RepoURL, pullRequestNumber int, mergeMethod gitprovider.MergeMethod, commitMesage string) error {
```

Comment:
What's the point of this parameter `mergeMethod` if it's always Merge?

Reply:
That's true, the alternative would be to `squash` merge but this func is only ever used to just merge 1 commit, so you're right that there's no need for `mergeMethod` - removed it in the latest commit!